src/core/config.rs

```diff
@@ -0,0 +1,135 @@
+use crate::core::Command;
+use termion::event::{Event, Key};
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use std::path::{Path, PathBuf};
+use std::fs;
+use std::collections::HashMap;
+
+use std::str::FromStr;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Keybinding {
+    keys: Vec<String>,
+    command: String,
+    // For now, unstructured value
+    args: Option<Value>,
+    context: Option<Value>,
+}
+
+pub struct KeybindingConfig {
+	pub keymap: HashMap<Event, Command>,
+	pub config_path: PathBuf
+}
+
+impl KeybindingConfig {
+	pub fn parse(config_path: &Path) -> Result<KeybindingConfig, Box<std::error::Error + Sync + Send + 'static>> {
+        let entries = fs::read_to_string(config_path)?;
+	    // Read the JSON contents of the file as an instance of `User`.
+	    let bindings: Vec<Keybinding> = json5::from_str(&entries)?;
+    	error!("Bindings parsed!");
+
+    	let mut keymap = HashMap::new();
+    	let mut found_cmds = Vec::new();
+    	for binding in bindings {
+    		let cmd = match Command::from_str(&binding.command) {
+    			Ok(cmd) => cmd,
+    			// unimplemented command for now
+    			Err(_) => continue,
+    		};
+    		error!("{:?} = {:?}", cmd, binding.keys);
+    		if found_cmds.contains(&cmd) {
+    			continue;
+    		}
+    		let binding = KeybindingConfig::parse_keys(&binding.keys).ok_or("Could not parse keybindings - config")?;
+    		keymap.insert(binding, cmd.clone());
+    		found_cmds.push(cmd);
+    	}
+
+    	Ok(KeybindingConfig{keymap: keymap, config_path: config_path.to_owned()})
+	}
+
+	fn parse_keys(keys: &Vec<String>) -> Option<Event> {
```

Comment:
In principle, yes. It means the key-text is unparsable. So there would only be one Err-condition. Option seemed easier/faster, just because of this simple "either we can parse it or we don't" distinction.